R/teal_data-extract.R

```diff
@@ -0,0 +1,38 @@
+#'
+#' @section Subsetting:
+#' `x[names]` subsets objects in `teal_data` environment and limit the code to the necessary needed to build limited
+#' objects.
+#'
+#' @param names (`character`) names of objects included in `teal_subset` to subset
+#' @param x (`teal_data`)
+#'
+#' @examples
+#'
+#' # Subsetting
+#' data <- teal_data()
+#' data <- eval_code(data, "a <- 1;b<-2")
+#' data["a"]
+#' data[c("a", "b")]
+#'
+#' join_keys(data) <- join_keys(join_key("a", "b", "x"))
+#' join_keys(data["a"]) # should show empty keys
+#' join_keys(data["b"])
+#' join_keys(data)["a"] # should show empty keys
+#' join_keys(data)["b"]
+#'
+#' @rdname teal_data
```

Comment:
Good catch!

Reply:
I googled a bit about `@describeIn` in https://roxygen2.r-lib.org/articles/reuse.html and it looks like it's not recommended and rather designed for a set of functions with the same set of arguments. I would rather stick to what we have right now as it lists S3 method on top of the `@usage` section. 

> In general, I no longer recommend @describeIn because I don’t think the heuristics it uses are as good as a thoughtful hand-crafted summary. If you’re currently using @describeIn, you can general replace it with @rdname, as long as you give some thought to the multiple-function @description.

I think the name of the manual page and the order of functions in `@usage` are taken from the order of files names.
So to have `teal_data` (file `teal_data.R`) to be before `[.teal_data` (teal_data-extract.R file) I needed to rename `teal_data.R` to `teal_data-constructor.R` so it comes before `teal_data-extract.R`. 

Check out the current order of functions and the name of the documentation c01612fdcc780e65984e6c10d2f285bd42041edc
<img width="455" alt="image" src="https://github.com/user-attachments/assets/29e85181-1858-45bf-a8ca-ae6c5dc13f37">

Reply:
Nice that the name was fixed. Yes, the Collate field of the DESCRIPTION is precisely whish R files go before and affects also the documentation. The `@describeIn` was a easy suggestion to move the documentation.